### DISCUSSION TITLE
The same but for natural

### DISCUSSION THREAD

**Discussion by yanivz** [IBM/zopeneditor-about#171] [General]
Adabas natural is a mainframe language, its looks similar to COBOL. Can I use this extention for this language?
Can IBM extend vs code extension to work with natural?

**Comment by FALLAI-Denis** [comment]
Does IBM provide a compiler for the Natural language?

I don't believe, so why should an IBM extension support a programming language not supported by IBM?
After search, the Natural language is a solution from Software AG. Rather, it would be up to them to provide a VS Code extension supporting their programming language.

In addition, the Z Open Editor extension, a free of charge solution, seems to me to be a subset of IBM Developer for Z, a paid solution. For a language to be supported by Z Open Editor, I think it must already be supported by IBM Developer for Z.

If the Natural language is not too complex, you could already create yourself a VS Code extension using a grammar (textmate format file) to provide syntax highlighting on source files opened in VS Code.

Ideally you should have a Language Server supporting Natural programming language, but developing such a component is difficult.
Software AG offers a solution under Eclipse: NaturalOne.
If they have developed a Language Server, they could integrate it into a VS Code extension.
This would be in their interest because we see that Eclise IDEs are in decline and that VS Code is becoming more and more widespread.
